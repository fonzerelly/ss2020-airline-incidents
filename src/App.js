import React from 'react';
import logo from './logo.svg';
import './App.css';
import Dummy from './Dummy'
import MyChart from './MyChart'
import ChartJsExample from './ChartJSExample';

function App() {
  return (
    <div>
          <Dummy/>
          {/* <MyChart/> */}
          <ChartJsExample/>
    </div>
  );
}

export default App;
