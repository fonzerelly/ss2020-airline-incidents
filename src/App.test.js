import React from 'react';
import { render, act } from '@testing-library/react';
import App from './App';
import AirlineSafetyContext from './airline-context';
import { readFileSync } from 'fs';
import parse from 'csv-parse/lib/sync'

describe('App', () => {

  it('should render as always without data', async () => {
      const tree = (
        <AirlineSafetyContext.Provider value={[[]]}>
        <App />
      </AirlineSafetyContext.Provider>
    )
    const { container } = render(tree);
    expect(container).toMatchSnapshot();
  });


  it('should render as always with data', async () => {
    const rawCsv = readFileSync(__dirname + '/../public/airline-safety.csv', 'utf8');
    const json = parse(rawCsv, {columns: true});
    const tree = (
      <AirlineSafetyContext.Provider value={[json]}>
        <App />
      </AirlineSafetyContext.Provider>
    )
    const { container } = render(tree);
    expect(container).toMatchSnapshot();
  });
})
