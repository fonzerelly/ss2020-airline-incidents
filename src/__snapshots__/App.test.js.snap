// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render as always with data 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      <h1>
        First Airline: 
        "Aer Lingus"
      </h1>
    </header>
  </div>
</div>
`;

exports[`App should render as always without data 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      <h1>
        First Airline: 
        none
      </h1>
    </header>
  </div>
</div>
`;
